File: java-suite/src/main/java/de/fraunhofer/iem/classesAndObjects/functionalInterface/Vehicle.java
package de.fraunhofer.iem.classesAndObjects.functionalInterface;

/**
 * Demonstrates the functional interface
 *
 * @author Ranjith Krishnamurthy
 */
@FunctionalInterface
public interface Vehicle {
    public void start();

    default public void stop() {
        System.out.println("Vehicle is stopped");
    }
}

File: kotlin-suite/src/main/kotlin/de/fraunhofer/iem/classesAndObjects/functionalInterface/Vehicle.kt
package de.fraunhofer.iem.classesAndObjects.functionalInterface

/**
 * Demonstrates the functional interface (SAM)
 *
 * @author Ranjith Krishnamurthy
 */
fun interface Vehicle {
    fun start()
    fun stop() {
        println("Vehicle is stopped")
    }
}